plugin/src/main/java/org/opensearch/ml/utils/error/ErrorMessage.java

```diff
@@ -50,14 +51,36 @@ protected String fetchReason() {
     }
 
     protected String fetchDetails() {
-        // Some exception prints internal information (full class name) which is security concern
-        return emptyStringIfNull(exception.getLocalizedMessage());
+        String msg;
+        // Prevent the method from exposing internal information such as internal ip address etc. that is a security concern.
+        if (hasInternalInformation(exception)) {
+            msg = decorateMessage(exception);
+        } else {
+            msg = exception.getLocalizedMessage();
+        }
+
+        return emptyStringIfNull(msg);
     }
 
     private String emptyStringIfNull(String str) {
         return str != null ? str : "";
     }
 
+    private Boolean hasInternalInformation(Throwable t) {
+        if (t instanceof ActionTransportException) {
```

Comment:
Is this exception coming from Core?

Reply:
Yes, any derived class from it has ip address.